test/unit-test/rtcp_api/rtcp_api_utest.c

```diff
@@ -635,170 +634,1295 @@ void test_rtcpDeSerializePacket( void )
+                       rtcpPacket.payloadLength );
+}
 
-    rtcpPacket.pPayload = NULL;
-
-    result = Rtcp_ParsePliPacket( &( context ),
-                                  &( rtcpPacket ),
-                                  &( rtcpPliPacket ) );
-
-    TEST_ASSERT_EQUAL( RTCP_RESULT_BAD_PARAM,
-                       result );
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate RTCP DeSerialize Packet functionality.
+ */
+void test_rtcpDeSerializePacket_TransportSpecificFeedback_TWCC( void )
+{
+    RtcpContext_t context;
+    uint8_t serializedPacket[] =
+    {
+        0x8F, 0xCD, 0x00, 0x0D,  /* Header: V=2, P=0, FMT=15, PT=RR=205, Length = 0xD words. */
+        0x87, 0x65, 0x43, 0x21, /* Sender SSRC */
+        /* Reception Report 1. */
+        0x00, 0x00, 0x00, 0x01, /* SSRC of first source. */
+        0x11, 0xA0, 0xA1, 0xA2, /* Fraction lost = 0x11, Cumulative packet lost = 0xA0A1A2. */
+        0xD1, 0xD2, 0xD3, 0xD4, /* Extended highest sequence number received = 0xD1D2D3D4. */
+        0xB1, 0xB2, 0xB3, 0xB4, /* Inter-arrival Jitter = 0xB1B2B3B4. */
+        0xC1, 0xC2, 0xC3, 0xC4, /* Last SR = 0xC1C2C3C4. */
+        0x5A, 0x5B, 0x5C, 0x5D, /* Delay since last SR = 0x5A5B5C5D. */
+        /* Reception Report 2. */
+        0x00, 0x00, 0x00, 0x02, /* SSRC of second source. */
+        0x11, 0xA0, 0xA1, 0xA2, /* Fraction lost = 0x11, Cumulative packet lost = 0xA0A1A2. */
+        0xD1, 0xD2, 0xD3, 0xD4, /* Extended highest sequence number received = 0xD1D2D3D4. */
+        0xB1, 0xB2, 0xB3, 0xB4, /* Inter-arrival Jitter = 0xB1B2B3B4. */
+        0xC1, 0xC2, 0xC3, 0xC4, /* Last SR = 0xC1C2C3C4. */
+        0x5A, 0x5B, 0x5C, 0x5D, /* Delay since last SR = 0x5A5B5C5D. */
+    };
+    size_t serializedPacketLength = sizeof( serializedPacket );
+    RtcpPacket_t rtcpPacket = { 0 };
+    RtcpResult_t result;
+
+    result = Rtcp_Init( &( context ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_OK,
+                       result );
+
+    result = Rtcp_DeserializePacket( &( context ),
+                                     &( serializedPacket[ 0 ] ),
+                                     serializedPacketLength,
+                                     &( rtcpPacket ) );
+
+    TEST_ASSERT_EQUAL( RTCP_RESULT_OK,
+                       result );
+    TEST_ASSERT_EQUAL( 0,
+                       rtcpPacket.header.padding );
+    TEST_ASSERT_EQUAL( 15,
+                       rtcpPacket.header.receptionReportCount );  /* Feedback message type (FMT) */
+    TEST_ASSERT_EQUAL_PTR( &( serializedPacket[ 4 ] ),
+                           rtcpPacket.pPayload );
+    TEST_ASSERT_EQUAL( serializedPacketLength - 4,
+                       rtcpPacket.payloadLength );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate RTCP DeSerialize Packet functionality.
+ */
+void test_rtcpDeSerializePacket_TransportSpecificFeedback_unknown( void )
```

Comment:
Similar question here, what part of this test case is unknown?

Reply:
Similarly unkown FMT in this case as well. Will write a comment here.